plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/CabalAdd.hs

```diff
@@ -0,0 +1,310 @@
+                                                                takeExtension,
+                                                                (</>))
+import           Text.PrettyPrint                              (render)
+import           Text.Regex.TDFA
+
+data Log
+  = LogFoundResponsibleCabalFile FilePath
+  | LogCalledCabalAddCodeAction
+  | LogCalledCabalAddCommand CabalAddCommandParams
+  | LogCreatedEdit WorkspaceEdit
+  | LogExecutedCommand
+  deriving (Show)
+
+instance Pretty Log where
+    pretty = \case
+      LogFoundResponsibleCabalFile fp -> "Located the responsible cabal file at " <+> pretty fp
+      LogCalledCabalAddCodeAction -> "The CabalAdd CodeAction is called"
+      LogCalledCabalAddCommand params -> "Called CabalAdd command with:\n" <+> pretty params
+      LogCreatedEdit edit -> "Created inplace edit:\n" <+> pretty edit
+      LogExecutedCommand -> "Executed CabalAdd command"
+
+cabalAddCommand :: IsString p => p
+cabalAddCommand = "cabalAdd"
+
+data CabalAddCommandParams =
+     CabalAddCommandParams { cabalPath   :: FilePath
+                           , verTxtDocId :: VersionedTextDocumentIdentifier
+                           , buildTarget :: Maybe String
+                           , dependency  :: T.Text
+                           , version     :: Maybe T.Text
+                           }
+  deriving (Generic, Show)
+  deriving anyclass (FromJSON, ToJSON)
+
+instance Pretty CabalAddCommandParams where
+  pretty CabalAddCommandParams{..} =
+    "CabalAdd parameters:" <+> vcat
+      [ "cabal path:" <+> pretty cabalPath
+      , "target:" <+> pretty buildTarget
+      , "dependendency:" <+> pretty dependency
+      , "version:" <+> pretty version
+      ]
+
+-- | Creates a code action that calls the `cabalAddCommand`,
+--   using dependency-version suggestion pairs as input.
+--
+--   Returns disabled action if no cabal files given.
+--
+--   Takes haskell file and cabal file paths to create a relative path
+--   to the haskell file, which is used to get a `BuildTarget`.
+--
+--   In current implementation the dependency is being added to the main found
+--   build target, but if there will be a way to get all build targets from a file
+--   it will be possible to support addition to a build target of choice.
+addDependencySuggestCodeAction
+  :: Recorder (WithPriority Log)
+  -> PluginId
+  -> VersionedTextDocumentIdentifier -- ^ Cabal's versioned text identifier
+  -> [(T.Text, T.Text)] -- ^ A dependency-version suggestion pairs
+  -> FilePath -- ^ Path to the haskell file (source of diagnostics)
+  -> FilePath -- ^ Path to the cabal file (that will be edited)
+  -> GenericPackageDescription
+  -> IO [CodeAction]
+addDependencySuggestCodeAction recorder plId verTxtDocId suggestions haskellFilePath cabalFilePath gpd = do
+    buildTargets <- liftIO $ getBuildTargets gpd cabalFilePath haskellFilePath
+    logWith recorder Info LogCalledCabalAddCodeAction
```

Comment:
Delete this kind of log message, I don't think it adds any value beyond this point.